pyfr/solvers/base/elements.py

```diff
@@ -211,47 +211,87 @@ def _gen_pnorm_fpts(self):
         self._norm_pnorm_fpts = pnorm_fpts / mag_pnorm_fpts[..., None]
         self._mag_pnorm_fpts = mag_pnorm_fpts
 
-    def _get_jac_eles_at(self, pts):
-        nspts, neles, ndims = self.nspts, self.neles, self.ndims
+    def _get_smats(self, pts, retdets=False):
         npts = len(pts)
 
-        # Form the Jacobian operator
-        jacop = np.rollaxis(self.basis.sbasis.jac_nodal_basis_at(pts), 2)
+        # Interpolation matrix to pts
+        M0 = self.basis.mbasis.nodal_basis_at(pts)
+
+        # Interpolate smats
+        smats = np.array([np.dot(M0, smat) for smat in self._smats])
+        smats = smats.reshape(self.ndims, npts, self.ndims, -1)
+
+        # Interpolate djacs
+        if retdets:
+            djacs = np.dot(M0, self._djacs)
+
+        return (smats, djacs) if retdets else smats
+
+    def _ploc_at_pts_np(self, pts):
+        op = self.basis.sbasis.nodal_basis_at(pts)
+
+        ploc = np.dot(op, self.eles.reshape(self.nspts, -1))
+        ploc = ploc.reshape(-1, self.neles, self.ndims).swapaxes(1, 2)
+
+        return ploc
+
+    @lazyprop
+    def _smats(self):
```

Comment:
Sorry for the confusing naming.
Now, get_smats() computes Smats by interpolation of Smats at pseudo grid points (_smats). 
How about renaming _smats as _smats_mpts? mpts denotes pseudo grid points.
Please let me know if you have a good naming for them.